docs: init technical roadmap (#219)
FuelLabs/docs-hub: docs/fuel-101/technical-roadmap.mdx

---
title: Technical Roadmap
category: Fuel 101
---

# Technical Roadmap

## Testnet Roadmap

> #### New abi encoding scheme

Reviewer: ```suggestion
#### New ABI encoding scheme
```

> The new abi encoding scheme will enable support for nested heap types.

Reviewer: ```suggestion
The new ABI encoding scheme enables support for nested heap types.
```

> This enables developers to return arbitrarily nested heap types from contracts and scripts, and log any arbitrary data type.

Reviewer: ```suggestion
This enables developers to return nested heap types from contracts, scripts, and log any arbitrary data type.
```

> A data type allocated on the heap can grow or shrink in size.
> Compared to the built-in array and tuple types which are allocated on the stack and cannot grow in serialize.

Reviewer: ```suggestion
A data type allocated on the heap can change in size, unlike built-in array and tuple types, which are allocated on the stack and remain fixed in size.
```

Reviewer: This reads much better and "cannot grow in serialize." doesn't make sense

#### Forkless upgrades in Fuel Core

> Supporting forkless upgrades to the fuel core will allow changing logic on the fly for all nodes in the network.

Reviewer: Can you explain how forkless upgrades work in 1 or 2 setences?

This minimizes the impact to the network when upgrades take place.

#### SRC-20 standardization for bridged assets

#### Fuel wallet connectors

Fuel wallet connectors enable users to sign transactions on the Fuel network using wallets such as Metamask or Phantom.

> These wallets are able to be used on Fuel through the use of predicates.

Reviewer: This is enabled through Fuel's support for multiple signatures can you talk briefly about which ones we support? Like sha256, keccak256, etc

#### The Graph integration

> Integrating Fuel with The Graph allows developers to easily index and query blockchain data for their Dapps.

Reviewer: ```suggestion
Integrating Fuel with The Graph allows developers to easily index and query blockchain data for their DApps.
```

Reviewer: Can close after this discussion
https://fuellabs.slack.com/archives/C060N9LTYG7/p1712860720233399?thread_ts=1712767928.815259&cid=C060N9LTYG7

> This integration will allow developers and users to easily aggregate, search, and filter the state of Dapps and the Fuel network.

Reviewer: ```suggestion
This integration will allow developers and users to easily aggregate, search, and filter the state of DApps and the Fuel network.
```

Reviewer: Can close after this discussion 
https://fuellabs.slack.com/archives/C060N9LTYG7/p1712860720233399?thread_ts=1712767928.815259&cid=C060N9LTYG7
